packages/outline-view/src/browser/outline-view-manager.ts

```diff
@@ -0,0 +1,50 @@
+/*
+* Copyright (C) 2017 TypeFox and others.
+*
+* Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+*/
+
+import { injectable } from "inversify";
+import { Event, Emitter } from "@theia/core";
+import { Range } from '@theia/editor/lib/browser';
+import URI from '@theia/core/lib/common/uri';
+
+@injectable()
+export class OutlineViewManager {
+
+    protected _isOpen: boolean = false;
+
+    protected readonly onDidChangeOutlineEmitter = new Emitter<OutlineSymbolInformation[]>();
+    protected readonly onDidChangeOpenStateEmitter = new Emitter<boolean>();
+
+    get onDidChangeOutline(): Event<OutlineSymbolInformation[]> {
+        return this.onDidChangeOutlineEmitter.event;
+    }
+
+    get onDidChangeOpenState(): Event<boolean> {
+        return this.onDidChangeOpenStateEmitter.event;
+    }
+
+    publish(symbolInformations: OutlineSymbolInformation[]): void {
+        this.onDidChangeOutlineEmitter.fire(symbolInformations);
+    }
+
+    get isOpen(): boolean {
+        return this._isOpen;
+    }
+
+    set isOpen(open: boolean) {
+        this._isOpen = open;
+        this.onDidChangeOpenStateEmitter.fire(open);
+    }
+}
+
+export interface OutlineSymbolInformation {
+    name: string;
+    id: string;
+    containerName?: string;
+    kind: string;
+    range: Range;
+    uri: URI;
```

Comment:
should contain `children: OutlineSymbolInformation[]`

Reply:
Also it should have callbacks for `onClick` and `onSelect` and the URI is not needed here. I want to be able to show trees without a URI.